### PR TITLE
feat: Add support for chart-releaser

### PR DESCRIPTION
**Pull request by NoUseFreak**
This allows users to publish to gh-pages using the cr tool.

### CODE REVIEW

File: lib/publish.js [hunk 1]
```diff
@@ -66,3 +66,65 @@ async function publishChart(configPath, registry, name, version) {
...
 }
+
+
+async function publishChartUsingCr(chartPath, crConfigPath, context) {
+    const logger = context.logger;
+    const env = context.env;
+
+    const crExec = await findCrExec()
+
+    const repoParts = /https:\/\/github.com(\/|:)([^/]+)\/([^\.]+)/.exec(context.options.repositoryUrl)
```
Comment (m1pl) [inline lib/publish.js:77]: Why not use `verifyCrGithubOnly` here?

File: lib/verifyConditions.js [hunk 2]
```diff
@@ -64,3 +83,36 @@ async function verifyS3Credentials(registryUrl) {
...
+
+async function verifyCrExecutable() {
+    try {
+        await execa('cr', ['version'])
+    } catch (error) {
+        const version = '1.2.1'
+        const cacheDir = '/tmp/cr/'
+        try {
+            await execa('mkdir', ['-p', cacheDir])
+            await execa('curl', ['-sSLo', 'cr.tar.gz', `https://github.com/helm/chart-releaser/releases/download/v${version}/chart-releaser_${version}_linux_amd64.tar.gz`])
```
Comment (m1pl) [inline lib/verifyConditions.js:108]: I don't think this tool should download and install other software. It should rather be provided through _Dockerfile_ or other tools that set up the CI environment.

Comment (NoUseFreak) [reply lib/verifyConditions.js:108]: I followed a similar part of the repo where it already installs a helm plugin.
https://github.com/m1pl/semantic-release-helm/blob/main/lib/verifyConditions.js#L54

File: lib/publish.js (hunk 1 above)
Comment (NoUseFreak) [reply lib/publish.js]: Updated

File: lib/verifyConditions.js (hunk 2 above)
Comment (m1pl) [reply lib/verifyConditions.js:113]: Ah right, forgot that part. It was from another PR, I don't know why I approved it like that ^^

**Pull request merged**